internal/util/secrets_test.go

```diff
@@ -15,50 +13,139 @@ package util
  limitations under the License.
 */
 
+package util
+
 import (
+	"errors"
 	"strings"
 	"testing"
+	"testing/iotest"
 	"unicode"
+
+	"gotest.tools/v3/assert"
+	"gotest.tools/v3/assert/cmp"
+	"k8s.io/apimachinery/pkg/util/sets"
 )
 
-func TestGeneratePassword(t *testing.T) {
-	// different lengths
-	for _, length := range []int{1, 2, 3, 5, 20, 200} {
-		password, err := GeneratePassword(length)
-		if err != nil {
-			t.Fatalf("expected no error, got %v", err)
-		}
-		if expected, actual := length, len(password); expected != actual {
-			t.Fatalf("expected length %v, got %v", expected, actual)
-		}
-		if i := strings.IndexFunc(password, func(r rune) bool { return !unicode.IsPrint(r) }); i > -1 {
-			t.Fatalf("expected only printable characters, got %q in %q", password[i], password)
-		}
-		if i := strings.IndexAny(password, passwordCharExclude); i > -1 {
-			t.Fatalf("expected no exclude characters, got %q in %q", password[i], password)
-		}
-	}
+func TestAccumulate(t *testing.T) {
+	called := 0
+	result, err := accumulate(10, func() (byte, error) {
+		called++
+		return byte('A' + called), nil
+	})
+
+	assert.NilError(t, err)
+	assert.Equal(t, called, 10)
+	assert.Equal(t, result, "BCDEFGHIJK")
 
-	// random contents
-	previous := []string{}
+	t.Run("Error", func(t *testing.T) {
+		called := 0
+		expected := errors.New("zap")
+		result, err := accumulate(10, func() (byte, error) {
+			called++
+			if called < 5 {
+				return byte('A' + called), nil
+			} else {
+				return 'Z', expected
+			}
+		})
+
+		assert.Equal(t, err, expected)
+		assert.Equal(t, called, 5, "expected an early return")
+		assert.Equal(t, result, "")
+	})
+}
+
+func TestGenerateAlphaNumericPassword(t *testing.T) {
+	for _, length := range []int{0, 1, 2, 3, 5, 20, 200} {
+		password, err := GenerateAlphaNumericPassword(length)
+
+		assert.NilError(t, err)
+		assert.Equal(t, length, len(password))
+		assert.Assert(t, cmp.Regexp(`^[A-Za-z0-9]*$`, password))
```

Comment:
:+1: I like this test. I think it does a reasonable job of checking the `AlphaNumberic` password generation feature, and it seems to be in line with what I'm seeing other places.